recipes/ada/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import get, copy, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+import os
+
+required_conan_version = ">=1.53.0"
+
+class AdaConan(ConanFile):
+    name = "ada"
+    description = "WHATWG-compliant URL parser written in modern C++"
+    license = ("Apache-2.0", "MIT")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ada-url/ada"
+    topics = ("url", "parser", "WHATWG")
+    package_type = "static-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "9",
+            "clang": "10",
+            "apple-clang": "12",
+            "Visual Studio": "16",
+            "msvc": "192",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        if self.settings.compiler == "clang" and Version(self.settings.compiler.version) < "12.0.0" and self.settings.compiler.libcxx != "libc++":
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires <charconv>, please use libc++ or upgrade compiler."
+            )
+
+    def _cmake_new_enough(self, required_version):
```

Comment:
I see this construction is used many times across recipes:
```
grep -RnI "def _cmake_new_enough" . | wc
     21      84    1929
```
maybe it's a time to open an issue in conan repo and ask to provide a helper out of the box?

Reply:
This is bad and should not be done :)